DEPR/CLN: Clean up to_dense signature deprecations

gfyoung opened the pull request:
Also deprecate `fill` in `.get_values` because its parameter was being passed to `.to_dense`, which
no longer accepts the `fill` parameter.

xref #14686.

pep8speaks commented:
Hello @gfyoung! Thanks for submitting the PR.

 - There are no PEP8 issues in the file [`pandas/core/sparse/array.py`](https://github.com/gfyoung/pandas/blob/21de4f49565831fe44f4c86a0ddad014e5c5a738/pandas/core/sparse/array.py) !

 - There are no PEP8 issues in the file [`pandas/core/sparse/series.py`](https://github.com/gfyoung/pandas/blob/21de4f49565831fe44f4c86a0ddad014e5c5a738/pandas/core/sparse/series.py) !

 - There are no PEP8 issues in the file [`pandas/tests/sparse/series/test_series.py`](https://github.com/gfyoung/pandas/blob/21de4f49565831fe44f4c86a0ddad014e5c5a738/pandas/tests/sparse/series/test_series.py) !

 - There are no PEP8 issues in the file [`pandas/tests/sparse/test_array.py`](https://github.com/gfyoung/pandas/blob/21de4f49565831fe44f4c86a0ddad014e5c5a738/pandas/tests/sparse/test_array.py) !

jreback commented:
hmm, this might need to go in after @TomAugspurger refactor of Sparse

gfyoung commented:
Sounds good.  I'll wait until that's done then.

gfyoung committed:
DEPR/CLN: Clean up to_dense signature deprecations

Also deprecate `fill` in `.get_values` because its
parameter was being passed to `.to_dense`, which
no longer accepts the `fill` parameter.

xref gh-14686.

gfyoung commented:
Looks like Azure is failing on `master` but could be addressed by #23182.

TomAugspurger commented:
I'm fine with ignoring these failures for now if the code isn't touching window / pyarrow.

gfyoung commented:
@TomAugspurger : These changes don't touch either of those.  That being said, I'm in no rush to merge these, and I see that you are getting close on #23182.  Unless somebody needs these deprecations removed *NOW*, I think I can hold.

jorisvandenbossche commented:
I think we could deprecate `get_values` all together? (instead of only deprecating the keyword)

gfyoung commented:
> I think we could deprecate get_values all together? (instead of only deprecating the keyword)

Ah, hence the `TODO` in the original code.  I think `to_dense` makes more sense semantically, so I don't see why we don't do that.  If there aren't any objections, I can do that as well in this PR.

jreback commented:
i agree we should deprecate get_values fully
but that means changing where it is used internally

i tried this a while back and it’s not so easy

maybe easier now that SpareArray is here

jorisvandenbossche commented:
"in principle" it should not be used internally (outside of sparse itself), as it is not part of the EA interface (or where it is used in sparse specific code, it should be replacable with to_dense) 
but, don't know to what extent this "in principle" is true :)

I think it is certainly worth to check in this PR.

gfyoung commented:
I haven't been following the sparse refactoring too closely, but to what extent are we trying to have a consistent interface for `SparseArray` with respect to `SparseSeries` and `SparseDataFrame` ?

I ask this because although I'm also inclined to deprecate `get_values`, it is of course a completely valid method for the latter two.  The devil's advocate in me thinks then that the interface for sparse object methods becomes a little inconsistent.

jreback commented:
this looks fine. can you rebase to get this passing. ping on green.

jreback commented:
I think we have an issue about deprecating ``.get_values()`` generally, which as I said above we should do (but slightly tricky).

jreback commented:
@gfyoung yep ok.

jorisvandenbossche commented:
> to what extent are we trying to have a consistent interface for SparseArray with respect to SparseSeries and SparseDataFrame ?

There does not need to be a consistency here, since we are considering deprecating SparseSeries/DataFrame alltogether.

I still think it doesn't make sense to only deprecate the feature if we are actually planning to deprecate the full method. I would directly do that, otherwise you annoy users twice for nothing.

TomAugspurger commented:
FYI, I've started a exploring a deprecation for SparseDataFrame /
SparseSeries. I think it'll be doable for 0.24.

On Wed, Oct 17, 2018 at 7:04 AM Joris Van den Bossche <
notifications@github.com> wrote:

> to what extent are we trying to have a consistent interface for
> SparseArray with respect to SparseSeries and SparseDataFrame ?
>
> There does not need to be a consistency here, since we are considering
> deprecating SparseSeries/DataFrame alltogether.
>
> I still think it doesn't make sense to only deprecate the feature if we
> are actually planning to deprecate the full method. I would directly do
> that, otherwise you annoy users twice for nothing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/22910#issuecomment-430601481>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABQHIuZgUn9b-20IRUSOIMMwCg4Xz1oKks5ulxzegaJpZM4XBGN8>
> .
>

gfyoung commented:
As @jreback had warned beforehand, there are some issues with trying to deprecate `to_dense` at the moment.  I would prefer to attempt to do this in a fresh PR once this is merged.

jorisvandenbossche commented:
I think it was about `get_values` and `is_dense` ? 

But, can you give a bit more details about the issues you encountered? That's useful information to keep track of.

gfyoung commented:
@jorisvandenbossche : There were some errors in the sparse index testing.  I'll try to debug.

gfyoung commented:
@jreback : Addressed all comments, and all is green again!  PTAL.

jorisvandenbossche commented:
@gfyoung was looking at the diff now, and was wondering: can't you simply remove the `fill` keyword in `get_values` as well instead of deprecating it? Because you already got a deprecation warning before, since this was simply passed to `to_dense`, which raised the future warning.

gfyoung commented:
@jorisvandenbossche : Ah!  That's a good point!  Let me that then.

gfyoung commented:
@jreback @jorisvandenbossche : Comments addressed, and all is green!  PTAL.

jreback commented:
thanks @gfyoung
